x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_indices.tsx

```diff
@@ -161,77 +168,92 @@ export const EngineIndices: React.FC = () => {
     },
   ];
   return (
-    <EnterpriseSearchEnginesPageTemplate
-      pageChrome={[engineName]}
-      pageViewTelemetry={EngineViewTabs.INDICES}
-      isLoading={isLoadingEngine}
-      pageHeader={{
-        pageTitle: i18n.translate('xpack.enterpriseSearch.content.engine.indices.pageTitle', {
-          defaultMessage: 'Indices',
-        }),
-        rightSideItems: [
-          <EuiButton data-test-subj="engine-add-new-indices-btn" iconType="plusInCircle" fill>
-            {i18n.translate('xpack.enterpriseSearch.content.engine.indices.addNewIndicesButton', {
-              defaultMessage: 'Add new indices',
-            })}
-          </EuiButton>,
-        ],
-      }}
-      engineName={engineName}
-    >
-      <>
-        <EuiInMemoryTable
-          items={indices}
-          columns={columns}
-          search={{
-            box: {
-              incremental: true,
-              placeholder: i18n.translate(
-                'xpack.enterpriseSearch.content.engine.indices.searchPlaceholder',
-                { defaultMessage: 'Filter indices' }
-              ),
-              schema: true,
-            },
-          }}
-          pagination
-          sorting
-        />
-        {removeIndexConfirm !== null && (
-          <EuiConfirmModal
-            onCancel={() => setConfirmRemoveIndex(null)}
-            onConfirm={() => {
-              removeIndexFromEngine(removeIndexConfirm);
-              setConfirmRemoveIndex(null);
+    <>
+      {isDeleteModalVisible ? (
+        <DeleteEngineModal engineName={engineName} onClose={closeDeleteEngineModal} />
```

Comment:
I don't think we need this here and in `engine_view.tsx` since `engine_view` renders the route that renders this component. Will this not render the `DeleteEngineModal` twice?

Reply:
good catch